src/adapt.c

```diff
@@ -54,6 +54,21 @@
     return ENOMEM; \
 }
 
+#define KNOBS_LOOP(_array, _index, _switch, _status, _cpu) \
```

Comment:
Are there different types involved here again? If not, can this be made a static function instead of a macro?